File: resources/information/technologies/c/specialization/AccountTest.c
#include "Account.h"
#include "ChequeAccount.h"

void processAccount(void * acc)
{
  printf("Processing account -> ");
  Account->print(acc);  /* done polymorphically */
  printf("\n");
  
  Account->credit(acc, 100); /* cheque accounts inherit credit */
  printf("After crediting with 100: "); 
  Account->print(acc);  /* done polymorphically */
  printf("\n");

  Account->debit(acc, 50);   /* done polymorphically */
  printf("After debiting with 50: ");
  Account->print(acc);  /* done polymorphically */
  printf("\n");
}

void testInheritanceAndPolymorphism()
{
   void * account = Account->create("Acc1");
   processAccount(account);   
   Account->delete((void * *)&account);  /* done polymorphically */
   
   account = ChequeAccount->create("ChAcc1", 6);
   processAccount(account);   
   Account->delete((void * *)&account);  /* done polymorphically */
}


int main()
{

  /* Testing inheritance */   
  
  testInheritanceAndPolymorphism(); 
  
  return 0;
};

File: resources/information/technologies/java/lang/classes/basic/classMembers/src/AccountTest.java
public class AccountTest
{
   public static void main(String[] args)
   {
     System.out.println("A: no of accounts = " + Account.getNumberOfInstances());
 
     Account a1 = new Account("A1");
     Account a2 = new Account("A2", 200);
 
     System.out.println("B: no of accounts = " + Account.getNumberOfInstances());
 
     a2 = null;
 
     System.out.println("C: no of accounts = " + Account.getNumberOfInstances());
 
     System.gc();
 
     System.out.println("D: no of accounts = " + Account.getNumberOfInstances());
 
   }
}
